pulsar-client-cpp/include/pulsar/KeyValue.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef KEY_VALUE_HPP_
+#define KEY_VALUE_HPP_
+
+#include <map>
+#include <string>
+#include <memory>
+#include "defines.h"
+#include "Schema.h"
+
+namespace pulsar {
+
+class KeyValueImpl;
+
+class PULSAR_PUBLIC KeyValue {
+   public:
+    KeyValue();
+    KeyValue(char *data, const int length, const KeyValueEncodingType &keyValueEncodingType);
```

Comment:
```suggestion
    KeyValue(const char *data, int length, const KeyValueEncodingType &keyValueEncodingType);
```

We usually don't add `const` to an integral parameter. Take the 4th overload of `std::string`'s [constructor](https://en.cppreference.com/w/cpp/string/basic_string/basic_string) as example:

```java
basic_string( const CharT* s, size_type count,
              const Allocator& alloc = Allocator() );
```

In addition, I think this constructor **should not be exposed to users**. It's only called in `Message::getKeyValueData`. We should compute the key and value and construct the `KeyValue` from these two strings.